src/BindExtensions.cs

```diff
@@ -22,13 +23,9 @@ public static (ValidationResult ValidationResult, T Data) BindAndValidate<T>(thi
                 data = Activator.CreateInstance<T>();
             }
 
-            var validatorType = request.HttpContext.RequestServices.GetService<IAssemblyProvider>()
-                .GetAssembly()
-                .GetTypes()
-                .FirstOrDefault(t => t.GetTypeInfo().BaseType != null &&
-                    t.GetTypeInfo().BaseType.GetTypeInfo().IsGenericType &&
-                    t.GetTypeInfo().BaseType.GetGenericTypeDefinition() == typeof(AbstractValidator<>) &&
-                    t.Name.Equals(typeof(T).Name + "Validator", StringComparison.OrdinalIgnoreCase));
+            var validators = request.HttpContext.RequestServices.GetService<IEnumerable<Type>>();
```

Comment:
I recommend making this a stronger contract. Storing types this loose in the DI system is a recipe for trouble. Make a real abstraction.

Reply:
Done! 😄